tests/realtikvtest/loaddatatest/load_data_test.go

```diff
@@ -849,3 +849,117 @@ func (s *mockGCSSuite) TestImportMode() {
 	s.Error(err)
 	s.Greater(intoNormalTime, intoImportTime)
 }
+
+func (s *mockGCSSuite) TestAddIndexBySQL() {
+	s.tk.MustExec("DROP DATABASE IF EXISTS load_data;")
+	s.tk.MustExec("CREATE DATABASE load_data;")
+	s.tk.MustExec(`CREATE TABLE load_data.add_index (a INT, b INT, c INT, PRIMARY KEY (a), unique key b(b), key c_1(c));`)
+
+	s.server.CreateObject(fakestorage.Object{
+		ObjectAttrs: fakestorage.ObjectAttrs{BucketName: "test-load", Name: "add_index-1.tsv"},
+		Content:     []byte("1,11,111\n2,22,222\n3,33,333\n4,44,444\n"),
+	})
+	s.server.CreateObject(fakestorage.Object{
+		ObjectAttrs: fakestorage.ObjectAttrs{BucketName: "test-load", Name: "add_index-2.tsv"},
+		Content:     []byte("5,55,555,\n6,66,666\n7,77,777"),
+	})
+	sql := fmt.Sprintf(`IMPORT INTO load_data.add_index FROM 'gs://test-load/add_index-*.tsv?endpoint=%s'`, gcsEndpoint)
+	s.tk.MustExec(sql)
+	s.tk.MustQuery("SELECT * FROM load_data.add_index;").Sort().Check(testkit.Rows(
+		"1 11 111",
+		"2 22 222",
+		"3 33 333",
+		"4 44 444",
+		"5 55 555",
+		"6 66 666",
+		"7 77 777",
+	))
+	s.tk.MustQuery("SHOW CREATE TABLE load_data.add_index;").Check(testkit.Rows(
+		"add_index CREATE TABLE `add_index` (\n" +
+			"  `a` int(11) NOT NULL,\n" +
+			"  `b` int(11) DEFAULT NULL,\n" +
+			"  `c` int(11) DEFAULT NULL,\n" +
+			"  PRIMARY KEY (`a`) /*T![clustered_index] CLUSTERED */,\n" +
+			"  UNIQUE KEY `b` (`b`),\n" +
+			"  KEY `c_1` (`c`)\n" +
+			") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin",
+	))
+
+	// encode error, rollback
+	s.tk.MustExec("truncate table load_data.add_index")
+	s.server.CreateObject(fakestorage.Object{
+		ObjectAttrs: fakestorage.ObjectAttrs{BucketName: "test-load", Name: "add_index-3.tsv"},
+		Content:     []byte("8,8|8,888\n"),
+	})
+	err := s.tk.ExecToErr(sql)
+	require.ErrorContains(s.T(), err, "Truncated incorrect DOUBLE value")
```

Comment:
need a case: use 3 engine, and fail on the third, check the table is truncated

Reply:
added

Reply:
check table empty?

Reply:
done